### ISSUE TITLE
Attachment path is not included in CSV import/export (I'm making a receipt program)

### ISSUE THREAD

**Issue by user8446** [moneymanagerex/moneymanagerex#2294]
When doing a CSV import or export, attachment paths are not included or supported. 

I'm making a receipt and invoice program that will do the OCR on receipts and invoices and put everything into the CSV ready to import into MMEX, eliminating all of the manually entry and manually attaching the receipt or invoice to the transaction.

 I can have the attachment automatically moved into the same MMEX archive folder and have the path put into the CSV so it can be directly opened in MMEX. As it is now I can only put the path into the notes section in which you would either have to copy/paste it into file explorer or manually navigate to it and open it to view it.

**Comment by user8446** [comment]
Hi @vomikan The BUG label should probably be added also as attachments should have already been included in CSV import

**Comment by vomikan** [comment]
@user8446 
In this sample I've added first available file name.
Question: what is the best solution here?

![image](https://user-images.githubusercontent.com/6836805/71755237-bb462880-2e9a-11ea-9f89-4b0bb3e5adc3.png)

**Comment by user8446** [comment]
@vomikan 
Hi, basically the attachment path would be an available field. That way, after import to view it in MMEX all that would need to be done is click the "open attachments". The only field to put it in now would be "notes":
![import screenshot](https://user-images.githubusercontent.com/18019245/71862263-4277e400-30c7-11ea-887b-4b91ffaa6e8c.png)

**Comment by vomikan** [comment]
I've added code to make attachments file path but all attachments has description

![image](https://user-images.githubusercontent.com/6836805/73136827-267eb700-4063-11ea-90da-974b47b5de30.png)

**Comment by user8446** [comment]
@vomikan 
That's great! If the attachment description will be a field then it can easily be populated with the payee, the word "receipt", the date, user defined, etc. on CSV import.
If not, a generic pre-populated title would even work such as "attachment".

**Comment by vomikan** [comment]
@user8446 

Returning to this question ....
I've developed an export to JSON file. This feature is available in v1.3.4 
In this format, the entirety of the data. There are no restrictions with the account currency, with transfers.
It is very difficult to make an adequate import from csv, given the diversity of formats.
JSON import not release yet.... To be developed.

**Comment by user8446** [comment]
@vomikan 
Thank you and looking forward to the JSON import! As it develops please have the attachment field available for import. A path or URL to the receipt location can be put there and users can easily click to instantly take them to the receipt image.

**Comment by vomikan** [comment]
@user8446 FYI 
WebApp has such functions like attachments import.
Look at this app too, it is very interesting. If more details are required, then I am ready to provide as ell as the author  @gabriele-v, I am sure.

**Comment by gabriele-v** [comment]
I agree with @vomikan, actually I think WebApp could be the closer thing that we have in MMEX to an Open API. If you are a developer and look at [MMEX Desktop WebApp integration code](https://github.com/moneymanagerex/moneymanagerex/blob/v1.3.5/src/webapp.cpp) and [WebApp itself](https://github.com/moneymanagerex/web-money-manager-ex/tree/master/WebApp) (mainly services.php and functions.php), you should be able to replicate an embedded webserver in your application and act as a "fake" WebApp. It should be much easier and "strict" than a CSV or JSON.

**Comment by user8446** [comment]
Thanks a bunch for the information... I will explore the WebApp
